src/NuGet.Core/NuGet.Frameworks/FrameworkNameProvider.cs

```diff
@@ -123,7 +123,29 @@ private static bool TryConvertOrNormalize(string key, IDictionary<string, string
             }
             else if (reverse.ContainsKey(key))
             {
-                value = reverse.Where(p => StringComparer.OrdinalIgnoreCase.Equals(p.Key, key)).Select(s => s.Key).Single();
+                bool found = false;
```

Comment:
I just realized this is getting a value from a dictionary, can't this just be:
```c#
else if (reverse.ContainsKey(key))
{
    foreach (KeyValuePair<string, string> item in reverse)
    {
        if (StringComparer.OrdinalIgnoreCase.Equals(item.Key, key))
        {
            value = item.Key;
            return true;
        }
    }
}
```

Reply:
Definitely! That's much cleaner. Added `NoAllocEnumerate()` as well to avoid the enumerator boxing where possible too.